tests/IsoCodes/Tests/SsnTest.php

```diff
@@ -78,14 +137,35 @@ public function testValidSsn($ssn)
     public function testInvalidSsn($ssn)
     {
         $this->assertFalse($this->ssn->validate($ssn));
+        $this->assertFalse(Ssn::validate($ssn));
+    }
+
+    /**
+     * run generate function to verify it not crash
+     *
+     * @param $state
+     *
+     * @dataProvider getStatesToGenerate
+     */
+    public function testGenerate($state)
+    {
+        $this->ssn->generate($state);
     }
 
     /**
-     * {@inheritdoc}
+     * Old constructor test. To be remove on 2.0.
      */
+    public function testOldConstructor()
+    {
+        \PHPUnit_Framework_Error_Deprecated::$enabled = false;
+        $ssn = new Ssn();
+        \PHPUnit_Framework_Error_Deprecated::$enabled = true;
+
+        $this->assertInstanceOf('IsoCodes\Ssn', $ssn);
+    }
+
     protected function setUp()
     {
-        parent::setUp();
-        $this->ssn = new Ssn();
+        $this->ssn = Ssn::getInstance();
```

Comment:
Just check code coverage. You will see both old and new system are tested and functional.

Better, test coverage increased on this PR.